yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -171,18 +173,20 @@ class MapOperator : public ObservableOperator<U, D> {
       : ObservableOperator<U, D>(std::move(upstream)),
         function_(std::move(function)) {}
 
-  void subscribe(Reference<Observer<D>> observer) override {
+  Reference<Subscription> subscribe(Reference<Observer<D>> observer) override {
+    auto subscription = make_ref<MapSubscription>(get_ref(this), std::move(observer));
     ObservableOperator<U, D>::upstream_->subscribe(
```

Comment:
Why don't we directly return the result of this call? Perhaps the ObservableOperator can augment the given subscription and return the augmented one. So, might be better to return that one.

```
return ObservableOperator<U, D>::upstream_->subscribe(make_ref<MapSubscription>(get_ref(this), std::move(observer));
```

Reply:
Funny you are asking something which I tried during working on this change. I added a unit test which shows we need to properly pass the cancel signal thru all chained operator subscriptions. The object dependencies between observable-subscriber-subscription is already so complicated that adding option to "augment" subscription feels to me like an overkill, not mentioning that its hard to ensure we have the right terminate signal chain.